proto/connect/trust_zone_service/v1alpha1/trust_zone_service.proto

```diff
@@ -26,6 +27,11 @@ message Agent {
   string trust_zone_id = 3;
 }
 
+message TrustZoneServer {
+  string cluster_id = 1;
+  string trust_zone_id = 2;
```

Comment:
This shouldn't strictly be necessary as we have the trust domain in the SPIFFE ID.

Reply:
We only have the trust zone ID in the context, not the cluster ID. So I think the cluster ID is necessary.